docs(telemetry): add tracing with the JS SDK guide (#46218)
supabase/supabase: apps/docs/content/guides/telemetry/client-side-tracing.mdx

---

> id: 'client-side-tracing'

Reviewer: Is there any intention of having tracing in other SDKs too? If so, we may want to rethink the structure later.

Author: Yes, there is intention to add it to the other SDKs!

Reviewer: Grand, then we can rethink the page later, possibly changing title etc

title: 'Tracing with the JS SDK'
description: 'Propagate W3C trace context from the Supabase JS SDK through Supabase services'
---

The Supabase JS SDK can attach [W3C Trace Context](https://www.w3.org/TR/trace-context/) headers (`traceparent`, `tracestate`, `baggage`) to outgoing requests. The resulting `trace_id` flows through Supabase services and appears in API Gateway and Edge Function logs, so you can correlate client-side spans with the server-side logs they produced — end-to-end, across the network boundary.

Because the headers follow the W3C standard, any compliant tracing SDK (OpenTelemetry, Sentry, Datadog, Honeycomb, etc.) can pick up the trace on the server side, including in self-hosted collectors.

## Requirements

- `@supabase/supabase-js` version `2.106.0` or later

> - `@opentelemetry/api` installed in your application

Reviewer 2: Worth noting this is often a transitive dependency — vendor SDKs like Sentry (v8+) and Datadog (`dd-trace`) already depend on `@opentelemetry/api`, so users of those SDKs don't need to install it separately. Maybe:

```md
- `@opentelemetry/api` resolvable at runtime (installed directly, or as a transitive
  dependency of your tracing SDK — e.g., Sentry v8+ and Datadog already include it)
```

Reviewer: ```suggestion
- `@opentelemetry/api` resolvable at runtime (installed directly, or as a transitive
  dependency of your tracing SDK — e.g., Sentry v8+ and Datadog already include it)
```

> - Any tracing SDK that emits W3C-compliant `traceparent` headers

Reviewer 2: This is slightly misleading, some vendor SDKs require explicit config for this. For example, Sentry's propagator only injects `traceparent` when `propagateTraceparent: true` is set in `Sentry.init()` — without it, only the proprietary `sentry-trace` header is injected and supabase-js silently gets nothing.

honestly this suggestion is really verbose, maybe there's a better way to add a vendor caveat section at the bottom of the doc, at least for agents to know they should check vendor docs if relevant

Suggested rewording:

```md
- A tracing SDK that registers a W3C-compliant propagator with the OpenTelemetry API.
  Vendor SDKs that wrap OTel (Sentry, Datadog, Honeycomb, etc.) work — but some require
  additional configuration to inject the standard `traceparent` header alongside their
  proprietary headers. Check your vendor's OTel integration docs.
```

Reviewer: ```suggestion
- A tracing SDK that registers a W3C-compliant propagator with the OpenTelemetry API.

  <Admonition type="note">
  
  Vendor SDKs that wrap OTel (Sentry, Datadog, Honeycomb, etc.) work — but some require additional configuration to inject the standard `traceparent` header alongside their proprietary headers. Check your vendor's OTel integration docs.
  
  </Admonition>
```

Formatting should be correct here…

If `@opentelemetry/api` is not installed, or there is no active trace context at request time, the SDK silently no-ops.

## Set up OpenTelemetry first

> The SDK reads from whatever `TracerProvider` is registered globally — it doesn't configure one for you. If you haven't instrumented your app yet, follow the [OpenTelemetry JavaScript getting started guide](https://opentelemetry.io/docs/languages/js/getting-started/) to install an SDK (`@opentelemetry/sdk-trace-node` for Node, `@opentelemetry/sdk-trace-web` for browsers) and an exporter for your backend (OTLP, Jaeger, Zipkin, or a vendor-specific one).

Reviewer: ```suggestion
The SDK reads from whatever `TracerProvider` you register globally — it doesn't configure one for you. If you haven't instrumented your app yet, follow the [OpenTelemetry JavaScript getting started guide](https://opentelemetry.io/docs/languages/js/getting-started/) to install an SDK (`@opentelemetry/sdk-trace-node` for Node, `@opentelemetry/sdk-trace-web` for browsers) and an exporter for your backend (OTLP, Jaeger, Zipkin, or a vendor-specific one).
```

Assuming it is the reader who configures it, makes the flow more consistent.

---
id: 'client-side-tracing'
title: 'Tracing with the JS SDK'
description: 'Propagate W3C trace context from the Supabase JS SDK through Supabase services'
---

The Supabase JS SDK can attach [W3C Trace Context](https://www.w3.org/TR/trace-context/) headers (`traceparent`, `tracestate`, `baggage`) to outgoing requests. The resulting `trace_id` flows through Supabase services and appears in API Gateway and Edge Function logs, so you can correlate client-side spans with the server-side logs they produced — end-to-end, across the network boundary.

Because the headers follow the W3C standard, any compliant tracing SDK (OpenTelemetry, Sentry, Datadog, Honeycomb, etc.) can pick up the trace on the server side, including in self-hosted collectors.

## Requirements

- `@supabase/supabase-js` version `2.106.0` or later
- `@opentelemetry/api` available at runtime — either installed directly or pulled in as a transitive dependency of your tracing SDK

> - A tracing SDK that registers a W3C-compliant propagator with the OpenTelemetry API

Reviewer 3: "using" might suit better in this context.

```suggestion
- A tracing SDK that registers a W3C-compliant propagator using the OpenTelemetry API
```

If `@opentelemetry/api` is not installed, or there is no active trace context at request time, the SDK silently no-ops.

> ## Set up OpenTelemetry first

Reviewer 3: If these headings and section rely on steps, then we could do:

```suggestion
## 1. Set up OpenTelemetry
```

To signal better these are sequential actions users need to make, and that their order is important.

The SDK reads from whatever `TracerProvider` you register globally — it doesn't configure one for you. If you haven't instrumented your app yet, follow the [OpenTelemetry JavaScript getting started guide](https://opentelemetry.io/docs/languages/js/getting-started/) to install an SDK (`@opentelemetry/sdk-trace-node` for Node, `@opentelemetry/sdk-trace-web` for browsers) and an exporter for your backend (OTLP, Jaeger, Zipkin, or a vendor-specific one).

The Supabase SDK only takes care of propagating the trace context that's already active when a request is made.

> 

Reviewer 3: I think we can make this shorter.

```suggestion 
The SDK reads the `TracerProvider` registerd globally as it doesn't provide one.

The Supabase SDK only propagates the trace context, already active when a request is made.

<Admonition>
If you haven't instrumented your app yet, follow the [OpenTelemetry JavaScript getting started guide](https://opentelemetry.io/docs/languages/js/getting-started/)
<Admonition>
```

Doesn't make sense to add steps of installing if we are already pointing them a guide.
